app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/LinkPreviewViewHolder.java

```diff
@@ -203,7 +203,7 @@ public void onClick(View view) {
             titleTextView.setVisibility(View.GONE);
             previewLayout.setVisibility(View.GONE);
 
-            setResponseListener(currContext);
+            responseListener = setResponseListener();
             RichPreview richPreview = new RichPreview(responseListener);
             responseListener = null;
```

Comment:
No need of this, what will setting this null do?

Reply:
@iamareebjamal since there was no method to stop the listener won't setting it null help in avoiding memory leaks ?

Reply:
No, There must be a method in richPreview to remove listener

Reply:
@iamareebjamal there is no method to remove the listener

Reply:
Done the changes @iamareebjamal